### PR TITLE
Add XCommon CMake support

### PR DESCRIPTION
**Pull request by danielpieczko**
- Added lib_build_info.cmake to provide XCommon CMake support
- Added an example application
- Added a Jenkinsfile

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  options {
+    buildDiscarder(xmosDiscardBuildSettings())
+    skipDefaultCheckout()
+    timestamps()
+  }
+  stages {
+    stage('Build') {
+      steps {
+        dir("lib_random") {
+          checkout scm
```
Comment (ed-xmos) [inline Jenkinsfile:21]: Lib checks missing?

**Review by ed-xmos — COMMENTED** [review]
I notice there is no test for this. Probably OK as quite a difficult thing to test.

https://softwareengineering.stackexchange.com/questions/147134/how-should-i-test-randomness

However given the scope of the current work backlog, this is probably out of scope.

Might be worth a quick review of the issues:

https://github.com/xmos/lib_random/issues

Some can be closed after your PR.

Also, missing readme for example.

File: Jenkinsfile (hunk 1 above)
Comment (danielpieczko) [reply Jenkinsfile:21]: There weren't any library checks in this repository, so they aren't being added at this point.

**Review by ed-xmos — APPROVED** [review]
Looks good

**Pull request merged**